torch/ao/quantization/_dbr/quantization_state.py

```diff
@@ -872,9 +909,25 @@ def _maybe_insert_output_observers(
         seen_q_op_info: SeenQOpInfo,
         root_module: torch.nn.Module,
     ):
-        func_output_obs_type = get_func_output_obs_type(seen_q_op_info)
-        output_tensor_id = seen_q_op_info.output_tensor_infos[0].id
+        is_start_of_fusion = seen_q_op_info.fusion_info and \
+            seen_q_op_info.fusion_info.is_first_element
+        is_not_start_of_fusion = seen_q_op_info.fusion_info and \
+            not seen_q_op_info.fusion_info.is_first_element
+
+        # if we are in a fusion but not at the start, do not insert observer
+        if is_not_start_of_fusion:
+            return
 
+        # if we are in a fusion and at the start, insert observer for its end
+        if is_start_of_fusion:
+            # get the output of the end of the fusion
+            cur_seen_q_op_info = get_seen_q_op_info_of_end_of_fusion(
+                seen_q_op_info, self.idx_to_seen_q_op_infos)
+            output_tensor_id = cur_seen_q_op_info.output_tensor_infos[0].id
+        else:
+            output_tensor_id = seen_q_op_info.output_tensor_infos[0].id
```

Comment:
nit: I feel restructuring the code to first check if there is fusion, then check if it is start of the fusion or not might be clearer. something like:
```
if seen_q_op_info.fusion_info:
    # in fusion
    # check if it's start or not
else:
   # not in fusion
```